pkg/etcdcli/health.go

```diff
@@ -81,12 +81,8 @@ func GetMemberHealth(ctx context.Context, cli *clientv3.Client, etcdMembers []*e
 }
 
 func checkSingleMemberHealth(ctx context.Context, cli *clientv3.Client, member *etcdserverpb.Member) healthCheck {
-	originalEndpoints := cli.Endpoints()
-	defer func(eps []string) {
-		cli.SetEndpoints(eps...)
-	}(originalEndpoints)
-
 	// we only want to check that one member that was passed, we're restoring the original client endpoints at the end
+	defer cli.SetEndpoints(cli.Endpoints()...)
```

Comment:
the actual problem is that the cli is shared amongst goroutines calling with different members in GetMemberHealth